rust/src/lib/nm/nm_dbus/nm_api.rs

```diff
@@ -435,6 +454,93 @@ impl NmApi<'_> {
     }
 }
 
+impl NmVersionInfo {
+    pub const CAPABILITY_SYNC_ROUTE_WITH_TABLE: usize = 0;
+
+    fn from(ver_info_arr: &[u32]) -> Self {
+        Self {
+            version_encoded: *ver_info_arr.first().unwrap_or(&0),
+            capabilities: ver_info_arr[1..].to_vec(),
+        }
+    }
+
+    pub fn version(&self) -> NmVersion {
+        NmVersion {
+            major: ((self.version_encoded >> 16) & 0xFF) as u8,
+            minor: ((self.version_encoded >> 8) & 0xFF) as u8,
+            micro: (self.version_encoded & 0xFF) as u8,
+        }
+    }
+
+    pub fn has_capability(&self, cap: usize) -> bool {
+        let idx = cap / 32;
+        let bit = cap % 32;
+        match self.capabilities.get(idx) {
+            Some(chunk) => (*chunk & (1 << bit)) != 0,
+            None => false,
+        }
+    }
+}
+
+impl NmVersion {
+    pub fn new(major: u8, minor: u8, micro: u8) -> NmVersion {
+        NmVersion {
+            major,
+            minor,
+            micro,
+        }
+    }
+
+    pub fn encoded(&self) -> u32 {
+        ((self.major as u32) << 16)
+            | ((self.minor as u32) << 8)
+            | (self.micro as u32)
+    }
+}
+
+impl Ord for NmVersion {
+    fn cmp(&self, other: &Self) -> std::cmp::Ordering {
+        self.encoded().cmp(&other.encoded())
```

Comment:
Optional to change:

If we have `From<NmVersion> for u32`, we can `u32::from(self).cmp(u32::from(other)`  which is easier to understand.

Reply:
I thought about implementing this automatic u32 <-> NmVersion conversions, but I thought that better not to do it. Not any u32 makes sense, it has a specific encoding that is not necessary obvious, so better to be explicit.